### PR TITLE
Implement sub_469D76

### PR DESCRIPTION
**Pull request by AaronVanGeffen**

### CODE REVIEW

File: src/OpenLoco/src/GameCommands/ClearLand.cpp [hunk 1]
```diff
@@ -11,17 +14,59 @@ namespace OpenLoco::GameCommands
...
-        call(0x00469D76, regs);
-        return regs.ebx;
+        call(0x004690FC, regs);
+    }
+
+    // 0x00469D76
+    static uint32_t sub_469D76(World::Pos2 pos, const uint8_t flags)
+    {
+        // This shoudn't happen due to using TilePosRangeView
+        if (pos.x > 0x2FFF || pos.y > 0x2FFF)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/GameCommands/ClearLand.cpp:36]: Do we have any constants for these yet?

Comment (duncanspumpkin) [reply src/OpenLoco/src/GameCommands/ClearLand.cpp:36]: that looks like a `validCoords`

File: src/OpenLoco/src/GameCommands/ClearLand.cpp [hunk 2]
```diff
@@ -11,17 +14,59 @@ namespace OpenLoco::GameCommands
...
+                sub_4690FC(pos);
+            }
+
+            auto tileHeight = World::TileManager::getHeight(pos);
+            TileManager::removeAllWallsOnTile(pos, tileHeight.landHeight / 4);
+        }
+
+        World::QuarterTile qt(0xF, 0);
+
+        // TODO: for _F0016C -- remove this hack when no longer needed
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/ClearLand.cpp:56]: Could you add a note that this is for `0x00469E07`

File: src/OpenLoco/src/GameCommands/ClearLand.cpp [hunk 3]
```diff
@@ -11,17 +14,59 @@ namespace OpenLoco::GameCommands
...
+    static loco_global<uint8_t*, 0x00F0016C> _F0016C; // stack offset to -eventually- flags
     static loco_global<uint16_t, 0x00F00170> _F00170;
+    static loco_global<uint32_t, 0x00F0013C> _F003CE; // tile x?
+    static loco_global<uint32_t, 0x00F00140> _F003D0; // tile y?
+    static loco_global<uint32_t, 0x00F25308> _F25308; // cost?
 
-    // 0x00469D76
-    static uint32_t sub_469D76(World::Pos2 pos, const uint8_t flags)
+    // 0x004690FC
+    static void sub_4690FC(World::Pos2 pos)
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/ClearLand.cpp:24]: can we give this a name `setTerrainStyleAsCleared`

File: src/OpenLoco/src/GameCommands/ClearLand.cpp [hunk 4]
```diff
@@ -11,17 +14,59 @@ namespace OpenLoco::GameCommands
...
         regs.ax = pos.x;
         regs.cx = pos.y;
-        regs.bl = flags;
-        call(0x00469D76, regs);
-        return regs.ebx;
+        call(0x004690FC, regs);
+    }
+
+    // 0x00469D76
+    static uint32_t sub_469D76(World::Pos2 pos, const uint8_t flags)
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/ClearLand.cpp:33]: `clearTile` ?

File: src/OpenLoco/src/GameCommands/ClearLand.cpp [hunk 5]
```diff
@@ -11,17 +14,59 @@ namespace OpenLoco::GameCommands
...
+            TileManager::removeAllWallsOnTile(pos, tileHeight.landHeight / 4);
+        }
+
+        World::QuarterTile qt(0xF, 0);
+
+        // TODO: for _F0016C -- remove this hack when no longer needed
+        uint8_t flagStackHack[10] = { 0 };
+        flagStackHack[8] = flags;
+
+        _F003CE = pos.x;
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/ClearLand.cpp:60]: Add comments that these are all part of 0x00469E07 parameters

File: src/OpenLoco/src/GameCommands/ClearLand.cpp [hunk 6]
```diff
@@ -11,17 +14,59 @@ namespace OpenLoco::GameCommands
...
+    static uint32_t sub_469D76(World::Pos2 pos, const uint8_t flags)
+    {
+        // This shoudn't happen due to using TilePosRangeView
+        if (!World::validCoords(pos))
+        {
+            GameCommands::setErrorText(StringIds::off_edge_of_map);
+            return GameCommands::FAILURE;
+        }
+
+        _F25308 = 0;
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/ClearLand.cpp:42]: should call this _sub469E07Cost or something make it clear what it is there for.

File: src/OpenLoco/src/Map/TileManager.cpp [hunk 7]
```diff
@@ -478,15 +480,36 @@ namespace OpenLoco::World::TileManager
...
         regs.cx = pos.y;
         regs.dl = baseZ;
         regs.dh = clearZ;
         regs.bl = qt.getBaseQuarterOccupied() | (qt.getZQuarterOccupied() << 4);
-        return !(call(0x00462926, regs) & Interop::X86_FLAG_CARRY);
+        return !(call(0x00462937, regs) & Interop::X86_FLAG_CARRY);
+    }
+
+    // 0x00462908
+    bool sub_462908(const World::Pos2& pos, uint8_t baseZ, uint8_t clearZ, const QuarterTile& qt, void* constructFunction)
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Map/TileManager.cpp:498]: i would call these clearFunctions not constructFunctions. They only ever remove things. (or do something else :))

File: src/OpenLoco/src/Map/TileManager.cpp [hunk 8]
```diff
@@ -478,15 +480,36 @@ namespace OpenLoco::World::TileManager
...
+    }
+
+    // 0x00462917
+    bool sub_462917(const World::Pos2& pos, uint8_t baseZ, uint8_t clearZ, const QuarterTile& qt, void* clearFunction)
+    {
+        return canConstructAt(pos, baseZ, clearZ, qt, clearFunction, (1 << 0));
+    }
+
+    // 0x00462926
+    bool canConstructAt(const World::Pos2& pos, uint8_t baseZ, uint8_t clearZ, const QuarterTile& qt, void* clearFunction)
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Map/TileManager.cpp:510]: in OpenRCT2 we called it the equivalent of canConstructAt if it didn't pass a clear function and canConstructWithClearAt if it did include a clear function. `0x00462926` is always one without a clear function.

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Map/TileManager.cpp:510]: Ahh, I see now. I'll remove the parameter for the outward-facing function, and rename the internal one, then.

File: src/OpenLoco/src/GameCommands/ClearLand.cpp [hunk 9]
```diff
@@ -11,17 +16,80 @@ namespace OpenLoco::GameCommands
...
+        if (surface->isIndustrial())
+        {
+            return;
+        }
+        if (surface->var_6_SLR5() > 0)
+        {
+            surface->setVar6SLR5(0);
+            surface->setVar4SLR5(0);
+
+            // originally a call to 0x4CC098
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/ClearLand.cpp:42]: we don't need these comments anymore.

File: src/OpenLoco/src/GameCommands/ClearLand.cpp [hunk 10]
```diff
@@ -11,17 +16,80 @@ namespace OpenLoco::GameCommands
...
+            Ui::ViewportManager::invalidate(pos, surface->baseHeight(), surface->baseHeight() + 32, ZoomLevel::eighth);
+        }
+        if (surface->var_4_E0() > 0)
+        {
+            surface->setVar4SLR5(0);
+
+            // originally a call to 0x4CC098
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/ClearLand.cpp:49]: same here

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**